src/core/pdf/base/XojCairoPdfExport.cpp

```diff
@@ -173,7 +183,11 @@ auto XojCairoPdfExport::createPdf(fs::path const& file, const PageRangeVector& r
         return false;
     }
 
-    if (!startPdf(file)) {
+    // Only export the outline if we are exporting the entire document. otherwise, links may point to page numbers
+    // greater than the total number of pages, leading to corrupt pdf files
+    bool outline = range.size() == 1 && range.front().first == 0 && range.front().last >= doc->getPageCount() - 1;
```

Comment:
This is only a sufficient condition for having all pages exported. If you export say a 3 page PDF with 
```xournalpp input.xopp -p export.pdf --export-range=1,2,3```
you get a different output (one without outline) than when you leave away the export-range.

This looks like a minor issue and maybe it's even wanted, but I'm pointing it out in case it wasn't noticed.